flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2706,6 +2717,78 @@ def test_remove_inode(self):
         )
 
 
+@implementer(IProfiledBlockDeviceAPI)
+class FakeProfiledLoopbackBlockDeviceAPI(
+        proxyForInterface(IBlockDeviceAPI, "_loopback_blockdevice_api")):
+    """
+    Fake implementation of ``IProfiledBlockDeviceAPI`` and ``IBlockDeviceAPI``
+    on top of ``LoopbackBlockDeviceAPI``. Profiles are not actually
+    implemented for loopback devices, but this fake is useful for testing the
+    intermediate layers.
+
+    :ivar _loopback_blockdevice_api: The underlying ``LoopbackBlockDeviceAPI``.
+    :ivar pmap dataset_profiles: A pmap from blockdevice_id to desired profile
+        at creation time.
+    """
+    def __init__(self, loopback_blockdevice_api):
+        self._loopback_blockdevice_api = loopback_blockdevice_api
+        self.dataset_profiles = pmap({})
+
+    def create_volume_with_profile(self, dataset_id, size, profile_name):
+        """
+        Calls the underlying ``create_volume`` on
+        ``_loopback_blockdevice_api``, but records the desired profile_name for
+        the purpose of test validation.
+        """
+        volume = self._loopback_blockdevice_api.create_volume(
+            dataset_id=dataset_id, size=size)
+        self.dataset_profiles = self.dataset_profiles.set(
+            volume.blockdevice_id, profile_name)
+        return volume
+
+
+def fakeprofiledloopbackblockdeviceapi_for_test(test_case,
+                                                allocation_unit=None):
+    """
+    Constructs a ``FakeProfiledLoopbackBlockDeviceAPI`` for use in tests that
+    want to verify functionality with an ``IProfiledBlockDeviceAPI`` provider.
+    """
+    return FakeProfiledLoopbackBlockDeviceAPI(
+        loopback_blockdevice_api=loopbackblockdeviceapi_for_test(
+            test_case, allocation_unit=allocation_unit))
+
+
+_fake_profiled_loopback_block_device_api = partial(
+    fakeprofiledloopbackblockdeviceapi_for_test,
+    allocation_unit=LOOPBACK_ALLOCATION_UNIT)
+
+
+class FakeProfiledLoopbackBlockDeviceIBlockDeviceTests(
```

Comment:
I guess there's a chance `FakeProfiledLoopbackBlockDeviceAPI` could break some aspect of `LoopbackBlockDeviceAPI`.

Doesn't seem terribly likely, and the former is test-only code so we generally verify that it is correct by using it in tests for something else and making sure the tests fail and pass in the ways we expect.  So I could definitely see dropping this, particularly if it adds noticably to the suite run time.